src/uwtools/drivers/chgres_cube.py

```diff
@@ -28,19 +28,31 @@ def namelist_file(self):
         yield self._taskname(f"namelist file {fn}")
         path = self._rundir / fn
         yield asset(path, path.is_file)
-        config_files = self._driver_config["namelist"]["update_values"]["config"]
-        input_paths = [
-            Path(config_files[k])
-            for k in ("mosaic_file_target_grid", "varmap_file", "vcoord_file_target_grid")
-        ] + [
-            Path(config_files["data_dir_input_grid"]) / config_files[k]
-            for k in ("atm_files_input_grid", "grib2_file_input_grid", "sfc_files_input_grid")
-            if k in config_files
-        ]
-        base_file = self._driver_config["namelist"].get("base_file")
-        yield [file(input_path) for input_path in input_paths] + (
-            [file(Path(base_file))] if base_file else []
-        )
+        input_files = []
+        namelist = self._driver_config["namelist"]
+        if base_file := namelist.get("base_file"):
+            input_files.append(base_file)
+        if update_values := namelist.get("update_values"):
+            config_files = update_values["config"]
+            for k in ["mosaic_file_target_grid", "varmap_file", "vcoord_file_target_grid"]:
+                input_files.append(config_files[k])
+            for k in [
+                "atm_core_files_input_grid",
+                "atm_files_input_grid",
+                "atm_tracer_files_input_grid",
+                "grib2_file_input_grid",
+                "nst_files_input_grid",
+                "orog_files_input_grid",
+                "orog_files_target_grid",
+                "sfc_files_input_grid",
+            ]:
+                if k in config_files:
+                    v = config_files[k]
+                    if isinstance(v, str):
+                        input_files.append(v)
+                    else:
+                        input_files += v
+        yield [file(Path(input_file)) for input_file in input_files]
```

Comment:
For files specified in the `chgres_cube` namelist, provide separate treatment for 1. Required single files (look up values with `[]` -- we expect the value to be available because the schema now requires it), and 2. Optional files that may be either single files or lists of files (look up with `.get()`).